dojo/utils.py

```diff
@@ -544,116 +641,60 @@ def add_breadcrumb(parent=None,
-                        start_of_week = start_of_next_week
-                        start_of_next_week += relativedelta(weeks=1)
-
-                    # finally a day that falls into the week bracket
-                    day_counts[day_offset[created.weekday()]] = day_count
-                    highest_day_count = max(highest_day_count, day_count)
-
-        # add week in progress + empty weeks on the end if needed
-        while tick < weeks + 1:
-            print(tick)
-            week_data, label = get_week_data(start_of_week, tick, day_counts)
-            print(week_data, label)
-            punchcard.extend(week_data)
-            ticks.append(label)
-            tick += 1
-
-            day_counts = [0, 0, 0, 0, 0, 0, 0]
-            start_of_week = start_of_next_week
-            start_of_next_week += relativedelta(weeks=1)
-
-        # adjust the size or circles
-        ratio = (sqrt(highest_day_count / pi))
-        for punch in punchcard:
-            # front-end needs both the count for the label and the ratios of the radii of the circles
-            punch.append(punch[2])
-            punch[2] = (sqrt(punch[2] / pi)) / ratio
-
-        return punchcard, ticks
-
-    except Exception as e:
-        logger.exception('Not showing punchcard graph due to exception gathering data', e)
-        return None, None
+def get_punchcard_data(findings, weeks_between, start_date):
+    punchcard = list()
+    ticks = list()
+    highest_count = 0
+    tick = 0
+    week_count = 1
+
+    # mon 0, tues 1, wed 2, thurs 3, fri 4, sat 5, sun 6
+    # sat 0, sun 6, mon 5, tue 4, wed 3, thur 2, fri 1
+    day_offset = {0: 5, 1: 4, 2: 3, 3: 2, 4: 1, 5: 0, 6: 6}
+    for x in range(-1, weeks_between):
+        # week starts the monday before
+        new_date = start_date + relativedelta(weeks=x, weekday=MO(1))
+        end_date = new_date + relativedelta(weeks=1)
+        append_tick = True
+        days = {0: 0, 1: 0, 2: 0, 3: 0, 4: 0, 5: 0, 6: 0}
+        for finding in findings:
+            try:
+                if new_date < datetime.combine(finding.date, datetime.min.time(
+                )).replace(tzinfo=timezone.get_current_timezone()) <= end_date:
+                    # [0,0,(20*.02)]
+                    # [week, day, weight]
+                    days[day_offset[finding.date.weekday()]] += 1
+                    if days[day_offset[finding.date.weekday()]] > highest_count:
+                        highest_count = days[day_offset[
+                            finding.date.weekday()]]
+            except:
+                if new_date < finding.date <= end_date:
+                    # [0,0,(20*.02)]
+                    # [week, day, weight]
+                    days[day_offset[finding.date.weekday()]] += 1
+                    if days[day_offset[finding.date.weekday()]] > highest_count:
+                        highest_count = days[day_offset[
+                            finding.date.weekday()]]
+                pass
```

Comment:
can you rebase your branch, looks like you are overwriting changes with older stuff